.github/workflows/main.yaml

```diff
@@ -23,15 +24,16 @@ jobs:
     strategy:
       matrix:
         platform: [ubuntu-latest]
-        node: [ '12', '10', '8', '6' ]
+        node: [12, 14, 16]
     name: CI - Node ${{ matrix.node }} (${{ matrix.platform }})
     runs-on: ${{ matrix.platform }}
     steps:
-      - uses: actions/checkout@v1
-      - uses: actions/setup-node@v1
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
         with:
           node-version: ${{ matrix.node }}
+      - run: npm i -g npm
       - name: Install dependencies
-        run: npm install --no-optional --no-save --no-update-notifier
+        run: npm ci
```

Comment:
Is there a good reason to use `npm ci` for a library? Normally this should be only used for applications. Lockfiles if packages are not published and ignored by npm.

Reply:
npm ci is slightly faster and installs the exact versions mentioned in the lockfile. since the lock file changes are undone. it won't work properly as expected. Will revert it.

Reply:
done.